gradle/publishMaven.gradle

```diff
@@ -7,14 +7,12 @@ ext {
 
 publishing {
   publications {
-    maven(MavenPublication) {
-      afterEvaluate {
-        pom.packaging = project.ext.packaging
-        if (project.ext.packaging != "pom") {
-          from(components.java)
-          artifact(sourcesJar)
-          artifact(javadocJar)
-        }
+    register("maven", MavenPublication) {
+      pom.packaging = project.ext.packaging
+      if (project.ext.packaging != "pom") {
+        from(components.java)
+        artifact(sourcesJar)
+        artifact(javadocJar)
```

Comment:
One more thing, the bom module needs `components.javaPlatform` instead of `components.java`, and `java-platform` plugin should be applied in bom modules.

Reply:
My fault, after applying db00f59d20eddca4b4c3ec62e72f6255f8eea41e & 0ec2232c8c40840d985c436dcb8f83007e3f5fa6, local run:

```sh
./gradlew publishToMavenLocal

tree ~/.m2/repository/org/spockframework/spock-bom/2.4-groovy-2.5-SNAPSHOT/
/Users/snebula/.m2/repository/org/spockframework/spock-bom/2.4-groovy-2.5-SNAPSHOT/
├── maven-metadata-local.xml
├── spock-bom-2.4-groovy-2.5-SNAPSHOT-javadoc.jar
├── spock-bom-2.4-groovy-2.5-SNAPSHOT-sources.jar
├── spock-bom-2.4-groovy-2.5-SNAPSHOT.jar
├── spock-bom-2.4-groovy-2.5-SNAPSHOT.module
└── spock-bom-2.4-groovy-2.5-SNAPSHOT.pom
```

Droping them for now, it looks normal.

```sh
tree ~/.m2/repository/org/spockframework/spock-bom/2.4-groovy-2.5-SNAPSHOT/
/Users/snebula/.m2/repository/org/spockframework/spock-bom/2.4-groovy-2.5-SNAPSHOT/
├── maven-metadata-local.xml
└── spock-bom-2.4-groovy-2.5-SNAPSHOT.pom
```